tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLAccessTest.java

```diff
@@ -215,27 +214,36 @@ public void testXMLNavigationExpressionWithQuotedIdentifiers() {
     @Test(groups = { "disableOnOldParser" })
     public void testInvalidXMLAccessWithIndex() {
         int i = 0;
-        BAssertUtil.validateError(negativeResult, i++, "cannot update an xml sequence", 5, 5);
-        BAssertUtil.validateError(negativeResult, i++, "invalid expr in assignment lhs", 13, 10);
-        BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found 'string'", 18, 15);
-        BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found 'boolean'", 19, 15);
-        BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found 'float'", 20, 15);
-
+        BAssertUtil.validateError(negativeResult, i++, "invalid expr in assignment lhs", 4, 10);
+        BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found 'string'", 9, 15);
+        BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found 'boolean'", 10, 15);
+        BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found 'float'", 11, 15);
+        BAssertUtil.validateError(negativeResult, i++, "cannot update an xml sequence", 18, 5);
+        BAssertUtil.validateError(negativeResult, i++, "invalid operation: type '(string|xml:Text)' does" +
+                " not support indexing", 21, 28);
+        BAssertUtil.validateError(negativeResult, i++, "invalid operation: type '(string|xml:Text)' does " +
+                "not support indexing", 22, 14);
+        BAssertUtil.validateError(negativeResult, i++, "invalid operation: union type '(xml:Element|xml<never> " +
```

Comment:
@suleka96 can you please create an issue to fix this error message, I don't think there should be `& readonly` part here as we didn't make this readonly.